DorisProject/src/app/pages/stock/stock.component.ts

```diff
@@ -66,8 +70,34 @@ export class StockComponent implements OnInit {
   }
 
   addProduct() {
-    this.router.navigate(['add-stock']);
+    this.router.navigate(['add-stock'], {state: {data: null}});
   }
 
+  edit(data: any) {
+    this.router.navigate(['add-stock'], {state: {data: data}});
+  }
+
+
+  openModal(request: string) {
+    const dialogRef = this.dialog.open(ModalComponent, {
+      width: '445px', height: '235px',
+      data: {textrequest: request, textresponse: this.responseModal}
```

Comment:
Codacy has a fix for the issue: [Replace `textrequest:·request,·textresponse:·this.responseModal}` with `·textrequest:·request,·textresponse:·this.responseModal·},`](https://app.codacy.com/gh/SaraArenasV/DorisProjectFrontEnd/pullRequest?prid=9258681)
```suggestion
      data: { textrequest: request, textresponse: this.responseModal },
```